### PR TITLE
Major plugin update including CVE fixes

### PR DESCRIPTION
**Pull request by bgermann**
This includes fixes for SECURITY-1014 (1) / CVE-2019-10471 (CSRF), CVE-2019-10472 as well as SECURITY-1014 (2) / CVE-2019-10473.
There are a lot of other improvements that make the code more robust and look better.
Additionally, the current wiki documentation is converted to live in git.

The libvirt-java bindings in released version 0.5.1 are not usable for JDK 8+ because they depend on an old JNA version. Therefore the dependency is marked as 0.5.2-SNAPSHOT because the current development version works.

### CODE REVIEW

**Comment by bgermann** [comment]
ping @magnayn @tastybug @mmornati

File: src/main/java/hudson/plugins/libvirt/BeforeJobSnapshotJobProperty.java [hunk 1]
```diff
@@ -54,7 +54,11 @@ public BeforeJobSnapshotJobProperty(EnableData snapshots) {
     }
 
     public String getSnapshotName() {
-        return getSnapshotsEnabled() ? snapshotsEnabled.getSnapshotName() : null;
+        if (!getSnapshotsEnabled()) {
+            return null;
```
Comment (mmornati) [inline src/main/java/hudson/plugins/libvirt/BeforeJobSnapshotJobProperty.java:58]: I'm always thinking that ternary operator is much simpler than the common if. But the result here is the same.

**Comment by bgermann** [comment]
@mmornati Would you support me adopting the plugin? I sent a [request for adoption](https://groups.google.com/d/msg/jenkinsci-dev/ncn0EAY-6J0/OhKDZ-HUAgAJ) to the dev mailing list, cc'ing the maintainers incl. you. Unfortunately, the mail sent to you was rejected.

**Comment by mmornati** [comment]
I guess I'm not maintainer anymore https://www.google.com/url?q=https%3A%2F%2Fgithub.com%2Fjenkins-infra%2Frepository-permissions-updater%2Fblob%2Fmaster%2Fpermissions%2Fplugin-libvirt-slave.yml&sa=D&sntz=1&usg=AFQjCNHbqKvxNNxrBAaHJMX-Uw1PqkoaJA
I already transfer it in the past. But no problem for me if there is a way to...

**Pull request merged**